enhancements/machine-api/cluster-api-integration.md

```diff
@@ -0,0 +1,256 @@
+---
+title: Cluster API Integration
+authors:
+  - "@JoelSpeed"
+  - "@alexander-demichev"
+reviewers:
+  - "@elmiko"
+  - "@Fedosin"
+  - "@lobziik"
+  - "asalkeld"
+  - "hardys"
+approvers:
+  - TBD
+creation-date: 2021-09-16
+last-updated: 2021-09-16
+status: implementable
+---
+
+# Cluster API Integration
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes the process of integrating the upstream [Cluster API](https://github.com/kubernetes-sigs/cluster-api)
+project into OpenShift standalone clusters.
+
+## Motivation
+
+We would like to give users the ablility to use Cluster API for machine management, as an alternative and possible future replacement for Machine API.
```

Comment:
should we perhaps say that we want to give users the ability to use CAPI as an _addition or supplement_ for machine management?

i don't think we are actually giving the ability to use CAPI as an alternative to MAPI at this point, or will a user be able to replace their initial MAPI MachineSets with CAPI MachineSets?

for example,

i am an OpenShift user, i do a normal install (now i have 3 Machines and 3 MachineSets managed by MAPI). i enable the CAPI tech preview and create 3 MachineSets each with 1 replica. can i now delete my original 3 worker nodes that are managed by MAPI?

Reply:
I find it a bit confusing that the description of the enhancement mentions `possible future replacement for Machine API` but then `Stop providing support for Machine API in near future` is stated as a non-Goal. I did notice that `near` in the non-Goal, which helps understanding the intention.

For the sake of scope and clarity, I would rather state that the intention is to replace MAPI in the future (basically without the `possible`) or just don't mention it at all. I lean more towards not mentioning the replacement at all, as that also aligns with the scenario metioned by @elmiko  (:wave: )

Reply:
You are right, I will not mention MAPI replacement in this proposal. I think the process of MAPI migration to CAPI might require a separate proposal.